src/oracle/connection/stmt.rs

```diff
@@ -17,13 +17,20 @@ pub struct Statement {
     buffers: Vec<Box<[u8]>>,
     sizes: Vec<i32>,
     indicators: Vec<Box<ffi::OCIInd>>,
+    pub mysql: String,
 }
 
 const NUM_ELEMENTS: usize = 20;
 
 impl Statement {
     pub fn prepare(raw_connection: &Rc<RawConnection>, sql: &str) -> QueryResult<Self> {
-        let mysql = sql.to_string();
+        let mut mysql = sql.to_string();
+        // TODO: this can go wrong: `UPDATE table SET k='LIMIT';`
+        if let Some(pos) = mysql.find("LIMIT") {
```

Comment:
Diesel does also support [offset](http://docs.diesel.rs/diesel/query_dsl/trait.QueryDsl.html#method.offset) so this should also be handled here.

@sgrif: Maybe that's something we could adjust in diesel 2.0? A solution would be to pack limit and offset together. So instead of the current `SelectStatement` something like this?
```rust
pub struct SelectStatement<
    From,
    Select = DefaultSelectClause,
    Distinct = NoDistinctClause,
    Where = NoWhereClause,
    Order = NoOrderClause,
    LimitOffset = NoLimitOffsetClause,
    GroupBy = NoGroupByClause,
    Locking = NoLockingClause,
> {
    pub(crate) select: Select,
    pub(crate) from: From,
    pub(crate) distinct: Distinct,
    pub(crate) where_clause: Where,
    pub(crate) order: Order,
    pub(crate) limitOffset: LimitOffset,
    pub(crate) offset: Offset,
    pub(crate) group_by: GroupBy,
    pub(crate) locking: Locking,
}
```

This would make it possible to implement that outside of diesel itself (given the re-rebalance coherence rfc)